### PR TITLE
docs(atomic): add storybook config for components requiring parent and child elements

### PR DESCRIPTION
**Pull request by olamothe**
Main changes are: 
* `default-story-shared.tsx`, `default-story.tsx` ,`default-result-component-story.tsx` now leverage two new story configuration property: `additionalChildMarkup` and `parentElement`
* `additionalChildMarkup` copies the other `*Markup` configuration option: It takes a lit html template, and add it as a child of the current atomic component story.
* `parentElement` allows to configure the container of an Atomic component (otherwise, default to search interface for standard component, and result list for result list elements). This is how, for example, when looking at `atomic-sort-expression` component, we can tell it to render inside a `atomic-sort` component.
* Segmented the stories using their title to show related components close to one another. For example, end users will see the `atomic-sort-expression` documentation as a child of `atomic-sort`, which does convey the fact that they need to be used in tandem.
- image
- image

When looking at the parent component documentation, you cannot modify the child components: Think of them as read only for the sake of the preview/live demo. And vice versa when looking at the documentation of a child component: You cannot modify the props of the parent element.

### CODE REVIEW

File: packages/atomic/src/components/formats/atomic-format-unit.result-number.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  'atomic-format-unit',
+  {
+    unit: 'byte',
+  },
+  FormatUnitDoc,
+  {
+    engineConfig: {
+      preprocessRequest: (r) => {
+        const parsed = JSON.parse(r.body as string);
+        parsed.fieldsToInclude = [...parsed.fieldsToInclude, 'size'];
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/formats/atomic-format-unit.result-number.stories.tsx:15]: isn't size a default field?

**Review by ThibodeauJF — APPROVED** [review]

**Review by npushkarskii — APPROVED** [review]

**Review by samisayegh — APPROVED** [review]

File: packages/atomic/.storybook/default-story-shared.tsx [hunk 2]
```diff
@@ -11,15 +11,32 @@ export const ADDON_PARAMETER_KEY = 'shadowParts';
 export interface DefaultStoryAdvancedConfig {
   engineConfig?: Partial<SearchEngineConfiguration>;
   additionalMarkup?: () => TemplateResult;
+  additionalChildMarkup?: () => TemplateResult;
+  parentElement?: () => HTMLElement;
```
Comment (btaillon) [inline packages/atomic/.storybook/default-story-shared.tsx:15]: It feels a bit weird to me to have three parameters so we can have children, parents and siblings. And it feels weirder to me that one of those takes an element directly, but not the others. One caveat with the `parentElement` element right now, for instance, is that we can't pass a deeper ancestry tree than 1.

Is there a way that we could support a config like the following, where one element would be replaced with the configurable element?
```jsx
// atomic-timeframe.stories.tsx
const {defaultModuleExport, exportedStory} = defaultStory(
  'Atomic/TimeframeFacet/Timeframe',
  'atomic-timeframe',
  {unit: 'year'},
  TimeframeDoc,
  {
    markup: html```
      <style>
        atomic-timeframe-facet {
          max-width: 500px;
          margin: auto;
        }
      </style>
      <atomic-timeframe-facet>
        <element /> <!-- This element would be automatically replaced with atomic-timeframe -->
      </atomic-timeframe-facet>
    ```
  }
);
```

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/formats/atomic-format-unit.result-number.stories.tsx (hunk 1 above)
Comment (olamothe) [reply packages/atomic/src/components/formats/atomic-format-unit.result-number.stories.tsx:15]: No, it's not !

File: packages/atomic/.storybook/default-story-shared.tsx (hunk 2 above)
Comment (olamothe) [reply packages/atomic/.storybook/default-story-shared.tsx:15]: That's not a bad idea honestly !

I'll file it under "to investigate" in JIRA

**Pull request merged**